2-1

TanVD opened the pull request.

TanVD committed:
Excersise 2-1. Main functionality is implemented, but i have a question about QTextStream and reason, why it is not working in more than one object for stdout

TanVD committed:
Few improvements and tests added

TanVD committed:
Few fixes

jzuken commented on 2_year/1_term/2-1/lanSurvey/OS.cpp:
```diff
@@ -0,0 +1,55 @@
+
+OS::OS(const QString &name)
+{
+    if (name.toLower() == "linux")
```
архитектура-шмархитектура, а потом всё равно пишем кучу if-ов. давайте лучше наследников сделаем этому классу

jzuken commented on 2_year/1_term/2-1/lanSurvey/RandomGenerator.cpp:
```diff
@@ -0,0 +1,18 @@
+    if (randomNumber < percent || percent == makeFakingPercent)
+    {
+        return true;
+    }
```
кто-то всё ещё так пишет. это так мило

jzuken commented on 2_year/1_term/2-1/lanSurvey/VirusModel.cpp:
```diff
@@ -0,0 +1,84 @@
+
+void VirusModel::createFromQStringList(const QStringList &list)
+{
+    foreach (QString name, list)
```
зачем же их копировать? по ссылкам надо итерироваться

jzuken commented on 2_year/1_term/2-1/lanSurvey/VirusModelTests.h:
```diff
@@ -0,0 +1,61 @@
+
+    void testForcedInfection()
+    {
+        RandomGenerator::makeFake(35);
```
хорошо, но не очень красиво. я бы сделал интерфейс генератора и две его реализации -- рандомную и фейковую. тогда в фейковую можно будет сразу пачкой последовательность значений загрузить

jzuken commented:
и что там с QTextStream?

TanVD commented:
Объекты QTextStream у меня в очередь писали в stdout и  записи следующего созданного по времени объекта выдавались только после смерти предыдущего, то есть они полность поток вывода перехватывали

TanVD committed:
1. Ifs replaced with hierarchy of classes inherited from OS.
2. Every OS class is created by OSSwithcer
3. Static implementation of RandomGenerator replaced with object.
4. Created Random generator interface and two implementations - random generator fake with list of fake percents and random generator - normal qrand generator
5. Few style fixes

jzuken commented:
ну как-то так, да
